### PR TITLE
10 implement rule set configuration

### PR DESCRIPTION
**Pull request by petermcd**
Implemented CPM parameter validation

## Summary by Sourcery

Implement a configurable rule-set framework by refactoring the base RuleSet interface and existing Logging/Prompts rules, and add a new CPMParameter rule set with lexer/parser/validator support and tests.

New Features:
- Introduce CPMParameter rule set to validate 'CPM Parameters Validation' sections and detect unused parameters
- Add new CPMParameterValidation token type in the lexer and corresponding dataclass

Enhancements:
- Refactor RuleSet base class to use abstract _get_config_key and _get_valid_token_types instead of static methods
- Update Logging and Prompts rule sets to leverage token objects for name/value checks and unified config interface
- Enhance logging validation logic to detect blank values, enforce lower-case, and recommend production defaults

Documentation:
- Fix spelling of 'behavior' to 'behaviour' in the bug report template

Tests:
- Add comprehensive tests for the CPMParameter rule set
- Add a new logging test case for empty value violation and standardize pytest.mark.parametrize formatting

### CODE REVIEW

File: tpc_plugin_validator/lexer/utilities/token_name.py [hunk 1]
```diff
@@ -8,5 +8,6 @@ class TokenName(Enum):
     ASSIGNMENT = 'ASSIGNMENT'
     COMMENT = 'COMMENT'
     FAIL_STATE = 'FAIL_STATE'
+    CPM_PARAMETER_VALIDATION = 'CPM_PARAMETER_VALIDATION',
```
[the remark this answers is not in the document]
Comment (petermcd) [reply tpc_plugin_validator/lexer/utilities/token_name.py:11]: Resolved in next commit.

File: tpc_plugin_validator/rule_sets/cpm_parameter.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        """
+        return 'cpm_parameter_validation'
+
+    def _get_valid_token_types(self) -> set[str]:
+        """
+        Provide a set of token types allowed in the section being analysed.
+
+        :return: Set of token types.
+        """
+        return {'Comment', 'CPM Parameter Validation',}
```
[the remark this answers is not in the document]
Comment (petermcd) [reply tpc_plugin_validator/rule_sets/cpm_parameter.py:70]: Raised as #23

File: tpc_plugin_validator/lexer/lexer.py [hunk 3]
```diff
@@ -28,16 +29,17 @@ def __init__(self, source: str) -> None:
         self._parsed_data: list[
             tuple[
                 TokenName,
-                Assignment | Comment | FailState | SectionHeader | StateTransition,
+                Assignment | Comment | FailState | CPMParameterValidation | SectionHeader | StateTransition,
             ]
         ] = []
         self._source = source
         self._token_specs = [
```
[the remark this answers is not in the document]
Comment (petermcd) [reply tpc_plugin_validator/lexer/lexer.py:36]: Raised as #24

**Comment by petermcd** [comment]
@sourcery-ai review

File: tpc_plugin_validator/rule_sets/cpm_parameter.py [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+                    description=f'The token type "{validation_token.token_name}" is not valid in the "CPM Parameters Validation" section, found on line {validation_token.line_number}.',
+                    severity=Severity.WARNING,
+                )
+                continue
+            if validation_token.token_name == 'Comment':
+                continue
+            if not self._check_parameter_usage(token=validation_token):
+                continue
+
+    def _check_parameter_usage(self, token: CPMParameterValidation):
```
[the remark this answers is not in the document]
Comment (petermcd) [reply tpc_plugin_validator/rule_sets/cpm_parameter.py:34]: Will implement at a later time

File: tpc_plugin_validator/rule_sets/rule_set.py [hunk 5]
```diff
@@ -52,11 +52,29 @@ def _add_violation(self, name: str, description: str, severity: Severity) -> Non
...
+        """
+        Check to ensure that the given token is valid for this section.
+
+        :param token: The token to check.
+
+        :return: True if valid, Otherwise False.
+        """
+        return token.token_name in self._get_valid_token_types()
+
+    @abstractmethod
```
[the remark this answers is not in the document]
Comment (petermcd) [reply tpc_plugin_validator/rule_sets/rule_set.py:65]: Not implementing at the moment.

File: tests/test_logging_rule_sets.py [hunk 6]
```diff
@@ -8,8 +8,10 @@
 
 
 class TestLoggingRuleSets(object):
+    """Test logging rule sets."""
+
     @pytest.mark.parametrize(
```
[the remark this answers is not in the document]
Comment (petermcd) [reply tests/test_logging_rule_sets.py:13]: Should be covered by existing checks.
